packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -3,32 +3,13 @@
  */
 import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Button } from '@wordpress/components';
-import { LEFT, RIGHT, UP, DOWN, BACKSPACE, ENTER } from '@wordpress/keycodes';
+import { Button, Notice } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import { URLInput } from '../';
 
-const handleLinkControlOnKeyDown = ( event ) => {
-	const { keyCode } = event;
-
-	if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( keyCode ) > -1 ) {
-		// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
-		event.stopPropagation();
-	}
-};
```

Comment:
While I don't love that we have to do this, I think removing this code might introduce an issue where the block toolbar will disappear when you use arrow keys.

Related: #19879

Reply:
Thank you for catching this.

I _could_ revert these commits

* https://github.com/WordPress/gutenberg/pull/19775/commits/b31be628e31e0287547546bfacdee75ed2f3fa54
* https://github.com/WordPress/gutenberg/pull/19775/commits/05e8e02ea57e2a56fb3085eda9b3143c8773ebe7

However, I don't entirely understand how these handlers ever work because I can't see how either prop is being used in the sub components. Note my comment on the commit msgs:

> This appears never to be called. The prop is not used directly within `URLInput` or “spread” from the passed props. Testing by adding console.log and running the tests confirms that the code is not being run. Manual testing also.

> Again onKeyDown prop is not used within `URLInput`. Nor is it spread from props. Never called in tests. Not called when testing manually either.

I've clearly not understood how this works. 

In addition, I think that if we do restore these handlers (and it seems we should?) then I need to add some very clear comments along with them to ensure they will never be removed again.

Reply:
We might want to test this. It could be working fine, in fact. I had forgotten that `URLInput` does its own `stopPropagation`:

https://github.com/WordPress/gutenberg/blob/0c089f7a481fae56dd493d679bf7b2ea000b21c4/packages/block-editor/src/components/url-input/index.js#L25-L28

>In addition, I think that if we do restore these handlers (and it seems we should?) then I need to add some very clear comments along with them to ensure they will never be removed again.

Well, ideally we _could_ remove `stopPropagation`, especially for the reasons outlined in #19879, but I do agree wholeheartedly with the idea of a comment which _cautions_ about why the code needs to exist and the conditions for its future removal.

Reply:
For me, it was just that neither of the key handling props being passed down into `URLInput` were being utilised within `URLInput`. I tried adding console logs just in case and they were never triggered.

What's the best way to test this?

Would it be relatively harmless to reinstate these handlers (by reverting previous commits as above) and create a new Issue to track this?

Reply:
> What's the best way to test this?

I expect it should be a matter of pressing one or more of these keys while within the LinkControl input, then observing whether or not the block toolbar disappears (a sign that the event is bubbling up to the ObserveTyping component, which we wouldn't want).

I tested it myself just now and it seems like it's working as well as it was before. As you mention, it's quite possible this was never really doing anything anyways.

> Would it be relatively harmless to reinstate these handlers (by reverting previous commits as above) and create a new Issue to track this?

It would be harmless and keep the pull request more focused, but based on the above, I also don't think it would be an issue to keep the changes as you've proposed them here.

Reply:
Ok let's keep the changes "as is". If something goes wrong at least we'll know why. I'd be surprised though cos those handlers were not used.

I also tested manually and found no bubbling I could observe.